build/build.py

```diff
@@ -499,10 +499,13 @@ def swigModule(self, **modArgs):
             # this turns the folder at the destination path into a package
 
             initTarget = init_tgen_name
-            bld(features = 'python_package',
-                name = initTarget,
-                target='__init__.py',
-                install_path = installPath)
+            try:
```

Comment:
OK cool - glad this worked.  This is going to be non-obvious when someone first looks at it (for example, in six months I will totally forget why we did it this way).  Can you add some comments explaining a) what get_tgen_by_name() is doing and why it's in a try/catch and b) what fundamentally we're doing here in terms of trying to get this target created exactly once but then reused across the targets with the same package name?